addons/hr_timesheet/models/project.py

```diff
@@ -223,8 +243,24 @@ def _fields_view_get(self, view_id=None, view_type='form', toolbar=False, submen
         """ Set the correct label for `unit_amount`, depending on company UoM """
         result = super(Task, self)._fields_view_get(view_id=view_id, view_type=view_type, toolbar=toolbar, submenu=submenu)
         result['arch'] = self.env['account.analytic.line']._apply_timesheet_label(result['arch'])
+
+        if view_type == 'tree' and self.env.company.timesheet_encode_uom_id == self.env.ref('uom.product_uom_day'):
+            result['arch'] = self._apply_time_label(result['arch'])
         return result
 
+    @api.model
+    def _apply_time_label(self, view_arch):
+        doc = etree.XML(view_arch)
+        encoding_uom = self.env.company.timesheet_encode_uom_id
+
+        for node in doc.xpath("//field[@name='planned_hours'][@widget='timesheet_uom'][not(@string)]"):
+            node.set('string', _('Planned %s') % encoding_uom.name or '')
+        for node in doc.xpath("//field[@name='effective_hours'][@widget='timesheet_uom'][not(@string)]"):
+            node.set('string', _('%s Spent') % encoding_uom.name or '')
+        for node in doc.xpath("//field[@name='remaining_hours'][@widget='timesheet_uom'][not(@string)]"):
+            node.set('string', _('Remaining %s') % encoding_uom.name or '')
+        return etree.tostring(doc, encoding='unicode')
```

Comment:
Could be great if it was managed client side, in the widget. But eh, it's already like that for the timesheet label :unamused: